### PR TITLE
Extending level_set/ex1 for initial visualization of the distorted si…

### PR DESCRIPTION
**Pull request by amneetb**
…gned distance field

### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [x] Was clang-format run?
- [ ] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [ ] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

**Comment by amneetb** [comment]
@Raamkrishnan : Can you add the volume calculation part in this example? Once that is done I (or David) will merge it with master as this one is just a simple extension of the existing example.

**Comment by drwells** [comment]
It looks like this got a few extra features past upgrading an example - what's the plan here?

**Comment by amneetb** [comment]
@Raamkrishnan : Please add the volume calculation part in this example, and then it should be ready to merge. 

@drwells : This is a straightforward extension of what was already in there. Should be good to merge once Rama adds a small piece of code to the example.

**Comment by amneetb** [comment]
This is good to go @drwells

File: src/level_set/fortran/levelsetops2d.f.m4
Comment (drwells) [inline src/level_set/fortran/levelsetops2d.f.m4:672]: same here - what's the difference between this one and `relaxationls3rdordereno2d` with `dir = 0`?

File: src/level_set/RelaxationLSMethod.cpp
Comment (drwells) [inline src/level_set/RelaxationLSMethod.cpp:795]: What's going on with `stage` here? This value isn't read. Could you add some documentation explaining what it might be intended to do in the future?

File: include/ibamr/RelaxationLSMethod.h
Comment (drwells) [inline include/ibamr/RelaxationLSMethod.h:166]: This isn't backwards-compatible - can you add a stub and deprecate it?

File: include/ibamr/ibamr_enums.h
Comment (drwells) [inline include/ibamr/ibamr_enums.h:221]: Can you add some documentation explaining what these do? Gauss-Seidel isn't normally thought of as a time integrator.

**Review by drwells — CHANGES_REQUESTED** [review]
I'm fine with merging example updates as-is but this adds new features so we need a test and a changelog entry.

**Comment by amneetb** [comment]
> I'm fine with merging example updates as-is but this adds new features so we need a test and a changelog entry.

I'll add one.

File: src/level_set/fortran/levelsetops2d.f.m4
Comment (amneetb) [reply src/level_set/fortran/levelsetops2d.f.m4:672]: Not for RK1 and RK2.

File: src/level_set/RelaxationLSMethod.cpp
Comment (amneetb) [reply src/level_set/RelaxationLSMethod.cpp:795]: I'll add that this could be used in the future to do RK3, etc.

File: include/ibamr/RelaxationLSMethod.h
Comment (amneetb) [reply include/ibamr/RelaxationLSMethod.h:166]: Is this about the `relax()` routine. This is private function, so it should not affect the compatibility, right?

File: src/level_set/fortran/levelsetops2d.f.m4
Comment (amneetb) [reply src/level_set/fortran/levelsetops2d.f.m4:456]: BTW, I can also remove the stage and just use dir = 0 and reuse the code. In the future, I can add it back if we ever want to do higher order RK integrators.

File: include/ibamr/ibamr_enums.h
Comment (boyceg) [reply include/ibamr/ibamr_enums.h:221]: Perhaps:
```suggestion
    GAUSS_SEIDEL_PSEUDO_TS = 1,
```
?

File: include/ibamr/RelaxationLSMethod.h
Comment (boyceg) [reply include/ibamr/RelaxationLSMethod.h:166]: I think it should be fine to make these kinds of changes to the private interface.

File: include/ibamr/RelaxationLSMethod.h
Comment (boyceg) [inline include/ibamr/RelaxationLSMethod.h:169]: Can you add a brief comment here about how G-S is being used as a pseudo-time integrator?

File: include/ibamr/RelaxationLSMethod.h
Comment (drwells) [reply include/ibamr/RelaxationLSMethod.h:166]: My mistake - I didn't see the `private:` label!

File: src/level_set/fortran/levelsetops2d.f.m4
Comment (drwells) [reply src/level_set/fortran/levelsetops2d.f.m4:456]: We've got a plan so I think implementing this with just a stub for now is OK.

File: include/ibamr/ibamr_enums.h
Comment (amneetb) [reply include/ibamr/ibamr_enums.h]: Did that!

File: include/ibamr/RelaxationLSMethod.h
Comment (amneetb) [reply include/ibamr/RelaxationLSMethod.h]: I added in here:

https://github.com/IBAMR/IBAMR/blob/amneetb/extending_LS_ex1/src/level_set/RelaxationLSMethod.cpp#L632-L650

**Comment by drwells** [comment]
There's a warning in the test output:
```
2023-05-24T19:55:54.1169296Z 1: Warning: The test(s) with input files 
2023-05-24T19:55:54.1169783Z 1: 
2023-05-24T19:55:54.1172862Z 1:     relaxation_ls_2d.input
2023-05-24T19:55:54.1174485Z 1: 
2023-05-24T19:55:54.1177766Z 1: cannot be run since either the input or output file missing. This usually means
2023-05-24T19:55:54.1179957Z 1: that one of the files is a broken link. Broken links to test files occur when
2023-05-24T19:55:54.1184062Z 1: a test exists in one revision of IBAMR but not another: for example, if a test
2023-05-24T19:55:54.1185058Z 1: has been removed then links to its input and output files may still be present
2023-05-24T19:55:54.1188460Z 1: in the build directory. This is usually harmless. A good way to get rid of this
2023-05-24T19:55:54.1191873Z 1: warning is to delete all test input and output files. To achieve this, execute
2023-05-24T19:55:54.1194954Z 1: the following commands in the build directory:
```
it looks like we have one serial input file and one parallel output file - can you add the missing files?

**Comment by amneetb** [comment]
> There's a warning in the test output:
> 
> ```
> 2023-05-24T19:55:54.1169296Z 1: Warning: The test(s) with input files 
> 2023-05-24T19:55:54.1169783Z 1: 
> 2023-05-24T19:55:54.1172862Z 1:     relaxation_ls_2d.input
> 2023-05-24T19:55:54.1174485Z 1: 
> 2023-05-24T19:55:54.1177766Z 1: cannot be run since either the input or output file missing. This usually means
> 2023-05-24T19:55:54.1179957Z 1: that one of the files is a broken link. Broken links to test files occur when
> 2023-05-24T19:55:54.1184062Z 1: a test exists in one revision of IBAMR but not another: for example, if a test
> 2023-05-24T19:55:54.1185058Z 1: has been removed then links to its input and output files may still be present
> 2023-05-24T19:55:54.1188460Z 1: in the build directory. This is usually harmless. A good way to get rid of this
> 2023-05-24T19:55:54.1191873Z 1: warning is to delete all test input and output files. To achieve this, execute
> 2023-05-24T19:55:54.1194954Z 1: the following commands in the build directory:
> ```
> 
> it looks like we have one serial input file and one parallel output file - can you add the missing files?

uh, made a mistake in naming the input file. Fixed and pushed it.

**Review by drwells — APPROVED** [review]

**Comment by drwells** [comment]
@boyceg are you satisfied with these updates? If so lets merge.

**Comment by amneetb** [comment]
> @boyceg are you satisfied with these updates? If so lets merge.

Ping :-)

File: include/ibamr/ibamr_enums.h
Comment (boyceg) [inline include/ibamr/ibamr_enums.h:232]: If RK1 is just forward Euler --- can we just list forward Euler?

File: include/ibamr/ibamr_enums.h
Comment (boyceg) [inline include/ibamr/ibamr_enums.h:234]: There is more than one RK2 --- could this be `EXPLICIT_MIDPOINT` or `EXPLICIT_TRAPEZOID` (or something else) as appropriate?

File: include/ibamr/ibamr_enums.h
Comment (boyceg) [inline include/ibamr/ibamr_enums.h:242]:
```suggestion
    if (val == GAUSS_SEIDEL_PSEUDO_TS) return "GAUSS_SEIDEL_PSEUDO";
```
?

**Review by boyceg — APPROVED** [review]
Looks good. I think that I would prefer to use more precise names than RK1 and RK2.

**Review by boyceg — APPROVED** [review]
Looks good. I think that I would prefer to use more precise names than RK1 and RK2.

File: include/ibamr/ibamr_enums.h
Comment (amneetb) [reply include/ibamr/ibamr_enums.h:234]: I am using TVD RK2. I that `EXPLICIT_TRAPEZOID`?

Comment (amneetb) [reply include/ibamr/ibamr_enums.h:234]: It is not `EXPLICIT_MIDPOINT` AFAICT from the top of my head.

File: include/ibamr/ibamr_enums.h
Comment (amneetb) [reply include/ibamr/ibamr_enums.h:232]: I thought of that, but then I thought to be consistent with other RK options it is better to write it as RK1. I don't have a strong opinion and can change it.

**Comment by amneetb** [comment]
I'll merge as all tests passed.

**Pull request merged**
